File: lab_2_02/cmake/graph_flex.cpp
#include <iostream>
#include <cmath>
#include <chrono>
#include <cstdlib>
#include <random>
#include <matplot/matplot.h>

using namespace std;

void fill_rand(int &len, int massive[][], int &upp_lim, int &low_lim){
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> distr(low_lim, upp_lim);
    for(int i=0; i < len; i++){
        massive[0][i] = distr(gen);
        massive[1][i] = distr(gen);
    };
}

float weight(int pos1[2], int pos2[2]){
    return sqrt(pow((pos1[1]-pos2[1]),2)+pow((pos1[2]-pos2[2]),2));
}

int main(){
    using namespace matplot;

    int len_sample = 10;
    int low_lim = 0;
    int upp_lim = 100;
    int coordinates[2][len_sample];
    fill_rand(len_sample, coordinates, upp_lim, low_lim);
    cout << "H";
    plot(,y);
}

File: lab_2_02/doc/graph_flex.py
from matplotlib import pyplot as plt
import numpy as np
import random as rd


def weight(pos1, pos2):
    return np.sqrt(pow((pos1[1] - pos2[1]), 2) + pow((pos1[2] - pos2[2]), 2))


def fill_rand(len, massive, upp_lim, low_lim):
    for i in range(len):
        massive[0][i] = rd.randint(low_lim, upp_lim)
        massive[1][i] = rd.randint(low_lim, upp_lim)
    return massive


len_sample = 10
low_lim = 0
upp_lim = 100
coordinates = []
coordinates.append([0]*len_sample)
coordinates.append([0]*len_sample)
coordinates = fill_rand(len_sample, coordinates, upp_lim, low_lim)
print(coordinates)